### PR TITLE
Export(package) support

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,8 @@ file(STRINGS "version.txt" __CPP_BINDGEN_VERSION)
 project(cpp_bindgen VERSION ${__CPP_BINDGEN_VERSION} LANGUAGES CXX)
 unset(__CPP_BINDGEN_VERSION)
 
+set(CMAKE_EXPORT_NO_PACKAGE_REGISTRY ON CACHE INTERNAL "")
```
Comment (lukasm91) [inline CMakeLists.txt:8]:
```suggestion
# Switch default of NO_PACKAGE_REGISTRY. TODO: Can be removed after CMake 3.15
set(CMAKE_EXPORT_NO_PACKAGE_REGISTRY ON CACHE INTERNAL "")
```

File: cmake/export.cmake [hunk 2]
```diff
@@ -1,36 +1,44 @@
...
-#     )
+# this registers the build-tree with a global CMake-registry
+export(PACKAGE cpp_bindgen)
 
 include(CMakePackageConfigHelpers)
 
-write_basic_package_version_file(cpp_bindgen-config-version.cmake
+# for build tree
+set(cpp_bindgen_MODULE_PATH ${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/build-install/lib/cmake)
+set(cpp_bindgen_SOURCES_PATH ${CMAKE_SOURCE_DIR}/src)
```
Comment (lukasm91) [inline cmake/export.cmake]: If c-bindings is added with add_subdirectory, will `CMAKE_SOURCE_DIR`, I would assume that `CMAKE_SOURCE_DIR` will point to the root source dir of the parent project? I have the feeling, this should be `PROJECT_SOURCE_DIR`.

Comment (lukasm91) [reply cmake/export.cmake]: same also at other places

**Review by lukasm91 — APPROVED** [review]
One comment, and one thing that you can also do in the next PR if you want... I didn't try it out, maybe you can do it, otherwise I'll probably try tomorrow.

File: cmake/export.cmake (hunk 2 above)
Comment (havogt) [reply cmake/export.cmake]: yes, you are right! thanks.

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Pull request merged**